tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/NUnit/CIVisibilityTestCommand.cs

```diff
@@ -23,73 +24,124 @@ public CIVisibilityTestCommand(ITestCommand innerCommand)
     [DuckReverseMethod]
     public object? Execute(object contextObject)
     {
+        Interlocked.CompareExchange(ref _totalRetries, CIVisibility.Settings.TotalFlakyRetryCount, -1);
         var context = contextObject.TryDuckCast<ITestExecutionContextWithRepeatCount>(out var contextWithRepeatCount) ?
                           contextWithRepeatCount :
                           contextObject.DuckCast<ITestExecutionContext>();
         var executionNumber = 0;
         var result = ExecuteTest(context, executionNumber++, out var isTestNew, out var duration);
-        if (result.ResultState.Status != TestStatus.Skipped &&
-            result.ResultState.Status != TestStatus.Inconclusive &&
-            isTestNew)
-        {
-            // Get retries number
-            var remainingRetries = Common.GetNumberOfExecutionsForDuration(duration) - 1;
+        var resultStatus = result.ResultState.Status;
 
-            // Retries
-            var retryNumber = 0;
-            var totalRetries = remainingRetries;
-            while (remainingRetries-- > 0)
+        if (resultStatus != TestStatus.Skipped &&
+            resultStatus != TestStatus.Inconclusive)
+        {
+            if (isTestNew)
```

Comment:
super-tiny nit, but I think it would be easier to follow here if you merged these two `if` and move the new `else if` "out".

Reply:
done something similar in [6cdb889](https://github.com/DataDog/dd-trace-dotnet/pull/6061/commits/6cdb8891eaeb28fba3851829ab29af7f5b57713c)